src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -209,6 +209,7 @@ public sealed partial class Server : IDisposable {
             Settings = newSettings;
 
             _symbolHierarchyMaxSymbols = Settings.analysis.symbolsHierarchyMaxSymbols;
+            _completionSource.Options = Settings.completion;
```

Comment:
`PythonCompletionOptions` is a class. Do we need to check `Settings.completion` for `null` here?

Reply:
It is poopulated in 
```cs
        [JsonRpcMethod("workspace/didChangeConfiguration")]
        public async Task DidChangeConfiguration(JToken token, CancellationToken cancellationToken) {
            using (await _prioritizer.ConfigurationPriorityAsync(cancellationToken)) {
                var settings = new LanguageServerSettings();

                var rootSection = token["settings"];
                var pythonSection = rootSection?["python"];
                if (pythonSection == null) {
                    return;
                }

                var autoComplete = pythonSection["autoComplete"];
                settings.completion.showAdvancedMembers = GetSetting(autoComplete, "showAdvancedMembers", true);
                settings.completion.addBrackets = GetSetting(autoComplete, "addBrackets", false);
```
and `GetSetting` handles nulls/missing settings.